modules/fedimint-wallet-client/src/lib.rs

```diff
@@ -634,6 +634,31 @@ impl WalletClientModule {
         Ok(self.module_api.module_consensus_version().await? >= ModuleConsensusVersion::new(2, 2))
     }
 
+    /// Allocates a deposit address controlled by the federation, guaranteeing
+    /// safe handling of all deposits, including on-chain transactions exceeding
+    /// `ALEPH_BFT_UNIT_BYTE_LIMIT`.
+    ///
+    /// Returns an error if the client is offline or the federation's wallet
+    /// module consensus version does not support processing all deposits.
+    pub async fn safe_allocate_deposit_address<M>(
+        &self,
+        extra_meta: M,
+    ) -> anyhow::Result<(OperationId, Address, TweakIdx)>
+    where
+        M: Serialize + MaybeSend + MaybeSync,
+    {
+        let module_consensus_version = self.module_api.module_consensus_version().await?;
```

Comment:
We have version discovery that includes consensus versions...

Ehhh... So much thought and infrastucture built to make things properly and then all of gets ignored to purse "simpler" approaches that work worse...

Reply:
Please correct me if you see otherwise, but I don't believe `discover_common_api_versions_set` is aware of the consensus module consensus version (voted module consensus). It's returning the hardcoded `MODULE_CONSENSUS_VERSION`.